src/core/prototypes/marks/icon.ts

```diff
@@ -344,4 +345,51 @@ export class IconElementClass extends EmphasizableMarkClass<
     ]);
     return widgets;
   }
+
+  public getTemplateParameters(): TemplateParameters {
+    const properties = [];
+
+    if (
+      this.object.mappings.visible &&
+      this.object.mappings.visible.type === "value"
```

Comment:
Not a big deal, but I don't remember if charticulator is using a newer version of typescript, but you can use optional chaining here if it does:

`if (this.object.mappings.visible?.type === "value")`

Reply:
It was introduced in TS 3.7.x. Charticulator uses 3.1.6.